workflows/funcscan.nf

```diff
@@ -97,54 +97,66 @@ workflow FUNCSCAN {
                         .mix(fasta_prep.uncompressed)
 
     // Some tools require annotated FASTAs
-    PROKKA ( ch_prepped_input, [], [] )
-    ch_versions = ch_versions.mix(PROKKA.out.versions)
-
+    // TODO only execute when we run tools that require prokka as input, e.g.
+    // if ( params.run_bgc_tool1 | params.run_bgc_tool2 | params.run_bgc_tool3 ) etc.
+    if ( params.run_arg_deeparg ) {
+        PROKKA ( ch_prepped_input, [], [] )
+        ch_versions = ch_versions.mix(PROKKA.out.versions)
+    }
     /*
         AMPs
     */
 
     // TODO AMPEP(?)
     // TODO ampir
-    MACREL_CONTIGS ( ch_prepped_input )
-    ch_versions = ch_versions.mix(MACREL_CONTIGS.out.versions)
-
+    if ( params.run_amp_macrel ) {
+        MACREL_CONTIGS ( ch_prepped_input )
```

Comment:
In the Macrel_Contigs option the tool uses Prodigal to predict genes. We should think of writing a module for Macrel_Peptides as well, in order not to run a gene prediction (with Prokka or Prodigal) two times during the pipeline.

Reply:
Good point! Can you make an issue?